pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -1005,7 +1075,7 @@ private int getOrCreateDocId() {
    */
   private IdMap<FixedIntArray> enableMetricsAggregationIfPossible(RealtimeSegmentConfig config,
```

Comment:
We can remove the return value and set both `_recordIdMap` and `_ingestionAggregator` in this method

Reply:
Because _recordIdMap is final, it doesn't like it being assigned within a method. I moved _ingestionAggregator from being a class member and instead have sourceColumn and valueAggregator as members of IndexContainer. Not touching the return type of this, but it all works (see code once I update PR).